### ISSUE TITLE
help issues?

### ISSUE THREAD

**Issue by rbonthond** [BrianPugh/cyclopts#215]
Not sure what I'm doing wrong here after upgrading to cyclopts from 2.4.2 to 2.9.5

```
from cyclopts import App, Parameter

app = App()

@app.default
def main(users: list[str] = ["a", "b"]) -> None:
    print(users)

if __name__ == "__main__":
    app()
```

running it like this:

```
$ python3 -V
Python 3.10.2
$ python3 testcase.py --help
Usage: main COMMAND [ARGS] [OPTIONS]

╭───────────────────── Traceback (most recent call last) ──────────────────────╮
│ example.py:11 in <module>                                            │
│                                                                              │
│    8 │   print(users)                                                        │
│    9                                                                         │
│   10 if __name__ == "__main__":                                              │
│ ❱ 11 │   app()                                                               │
│   12                                                                         │
│                                                                              │
│ /python3/lib/python3.10/site-packages/cyclopts/core.py:890 in    │
│ __call__                                                                     │
│                                                                              │
│    887 │   │   │   │                                                         │
│    888 │   │   │   │   return asyncio.run(command(*bound.args, **bound.kwarg │
│    889 │   │   │   else:                                                     │
│ ❱  890 │   │   │   │   return command(*bound.args, **bound.kwargs)           │
│    891 │   │   except Exception as e:                                        │
│    892 │   │   │   try:                                                      │
│    893 │   │   │   │   from pydantic import ValidationError as PydanticValid │
│                                                                              │
│ /python3/lib/python3.10/site-packages/cyclopts/core.py:954 in    │
│ help_print                                                                   │
│                                                                              │
│    951 │   │   help_format = resolve_help_format(apps)                       │
│    952 │   │   console.print(format_doc(self, executing_app, help_format))   │
│    953 │   │                                                                 │
│ ❱  954 │   │   for help_panel in self._assemble_help_panels(tokens, help_for │
│    955 │   │   │   console.print(help_panel)                                 │
│    956 │                                                                     │
│    957 │   def _resolve_command(                                             │
│                                                                              │
│ /python3/lib/python3.10/site-packages/cyclopts/core.py:1037 in   │
│ _assemble_help_panels                                                        │
│                                                                              │
│   1034 │   │   │   │   │   _, existing_panel = panels[group.name]            │
│   1035 │   │   │   │   except KeyError:                                      │
│   1036 │   │   │   │   │   existing_panel = None                             │
│ ❱ 1037 │   │   │   │   new_panel = create_parameter_help_panel(group, iparam │
│   1038 │   │   │   │                                                         │
│   1039 │   │   │   │   if existing_panel:                                    │
│   1040 │   │   │   │   │   # An imperfect merging process                    │
│                                                                              │
│ /python3/lib/python3.10/site-packages/cyclopts/help.py:353 in    │
│ create_parameter_help_panel                                                  │
│                                                                              │
│   350 │   │   │   help_append(rf"[env var: {env_vars}]", "dim")              │
│   351 │   │                                                                  │
│   352 │   │   if cparam.show_default or (                                    │
│ ❱ 353 │   │   │   cparam.show_default is None and iparam.default not in {Non │
│   354 │   │   ):                                                             │
│   355 │   │   │   default = ""                                               │
│   356 │   │   │   if isclass(type_) and issubclass(type_, Enum):             │
╰──────────────────────────────────────────────────────────────────────────────╯
TypeError: unhashable type: 'list'

```

**Comment by BrianPugh** [comment]
Most immediately, it's bad practice to assign a mutable default in a function signature:
```
def main(users: list[str] = ["a", "b"]) -> None:
    ...
```
The preferred alternative is (which works with cyclopts):
```
def main(users: Optional[list[str]] = None) -> None:
    if users is None:
        users = ["a", "b"]
```

Regardless, I'll investigate to see what's up.

**Comment by rbonthond** [comment]
fair enough, that is better python. Though I liked that in 2.4.2 the defaults showed up in the help

```
python3 example.py --help
Usage: main COMMAND [ARGS] [OPTIONS]

╭─ Commands ───────────────────────────────────────────────────────────────────╮
│ --help,-h  Display this message and exit.                                    │
│ --version  Display application version.                                      │
╰──────────────────────────────────────────────────────────────────────────────╯
╭─ Parameters ─────────────────────────────────────────────────────────────────╮
│ USERS,--users,--empty-users  [default: ['a', 'b']]                           │
╰──────────────────────────────────────────────────────────────────────────────╯
```

**Comment by BrianPugh** [comment]
This is fixed in #216, and I'll release v2.9.6 with this in a few minutes.
